File: task-03/elixir/Sub-Task 4.ex
IO.puts("Enter a number: ")
num = IO.gets("") |> String.trim() |> String.to_integer()
File.write!("input.txt", Integer.to_string(num))
{:ok, content} = File.read('input.txt')
rows = String.trim(content) |> String.to_integer()
pattern =
  for i <- 0..(rows - 2) do
    for _ <- 0..(rows - i - 1) do
      IO.write(" ")
    end
    for _ <- 0..i do
      IO.write("* ")
    end
    IO.puts("")
  end

  for i <- 0..(rows - 1) do
    # Print leading spaces
    for _ <- 0..i do
      IO.write(" ")
    end

    # Print asterisks
    for _ <- 0..(rows - i - 1) do
      IO.write("* ")
    end

    IO.puts("")
  end
File.write!("output.txt", pattern)

{:ok, new} = File.read("output.txt")
 IO.puts(new)

File: task-03/Javascript/Sub-Task 4.js
const fs = require('fs');
fs.readFile('input.txt', 'utf8', (err, data) => {
    if (err) throw err;
    const rows = parseInt(data.trim());
    let diamond = '';
    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < rows - i - 1; j++) {
            diamond += " ";
        }
        for (let j = 0; j <= i; j++) {
            diamond += "* ";
        }
        diamond += '\n';
    }
    for (let i = 0; i < rows - 1; i++) {
        for (let j = 0; j <= i; j++) {
            diamond += " ";
        }
        for (let j = 0; j < rows - i - 1; j++) {
            diamond += "* ";
        }
        diamond += '\n';
    }
    fs.writeFile('output.txt', diamond, (err) => {
        if (err) throw err;
        console.log('Diamond');
    });
});
